docs/authentication/jwt-auth.html

```diff
@@ -16,13 +16,407 @@
+<pre><code class="language-swift">import SwiftJWT
+import Kitura
+
+struct TypeSafeJWT&lt;C: Claims&gt;: TypeSafeMiddleware {
+    static func handle(request: RouterRequest, response: RouterResponse, completion: @escaping (TypeSafeJWT<C>?, RequestError?) -> Void) {
+
+    }
+}</code></pre>
+    <p class="block-text">The `handle` function is required for a TypeSafeMiddleware and is where we will interact with the headers.
+    We will the return an instance of TypeSafeJWT on success or a RequestError on failure. The struct is generic so it can be used on any JWT.</p>
+    <p>Within the handler we are interested in the decoded JWT so we add that as a field:</p>
+    <pre><code class="language-swift">let jwt: JWT<C></code></pre>
+    <p>Finally, we initialize our JWT using the same functions as our protected get route.</p>
+    <p>Inside our handle function, add the following:</p>
+    <pre><code class="language-swift">let authHeader = request.headers["Authorization"]
+guard let authComponents = authHeader?.components(separatedBy: " "),
+    authComponents.count == 2,
+    authComponents[0] == "Bearer",
+    let jwt = try? JWT&lt;C&gt;(jwtString: authComponents[1], verifier: App.jwtVerifier)
+else {
+    return completion(nil, .unauthorized)
+}
+completion(TypeSafeJWT(jwt: jwt), nil)</code></pre>
+<p>Your completed `TypeSafeJWT.swift` file should look as follows:</p>
+<pre><code class="language-swift">import SwiftJWT
+import Kitura
+
+struct TypeSafeJWT&lt;C: Claims&gt;: TypeSafeMiddleware {
+    let jwt: JWT&lt;C&gt;
+    static func handle(request: RouterRequest, response: RouterResponse, completion: @escaping (TypeSafeJWT<C>?, RequestError?) -> Void) {
+        let authHeader = request.headers["Authorization"]
+        guard let authComponents = authHeader?.components(separatedBy: " "),
+            authComponents.count == 2,
+            authComponents[0] == "Bearer",
+            let jwt = try? JWT&lt;C&gt;(jwtString: authComponents[1], verifier: App.jwtVerifier)
+        else {
+            return completion(nil, .unauthorized)
+        }
+        completion(TypeSafeJWT(jwt: jwt), nil)
+    }
+}</code></pre>
+<div class="underline"></div>
+
+<h2 class="heading-2"><span class="blue-text">Step 9b:</span> Register TypeSafeJWT on a route.</h2>
+    <p>Back in our `JWTRoutes.swift`, We are going to register a new route using this TypeSafeMiddleware.</p>
+    <p>In your `initializeJWTRoutes` add a new route:</p>
+    <pre><code class="language-swift">app.router.get("/jwtCodable", handler: app.typeSafeHandler)</code></pre>
+    <p>In your app extension, define the typeSafeHandler:</p>
+<pre><code class="language-swift">func typeSafeHandler(typeSafeJWT: TypeSafeJWT&lt;ClaimsStandardJWT&gt;, completion: (JWT&lt;ClaimsStandardJWT>?, RequestError?) -> Void) {
+    completion(typeSafeJWT.jwt, nil)
+}</code></pre>
+    <p>This will run your TypeSafeJWT middleware and on success return the jwt just like out raw route.</p>
```

Comment:
I think we should reword this as "and on success" sounds like reading code - "This function will run the middleware, `TypeSafeJWT`, and if it succeeds it will return the JWT, just as we did in our raw routing example above."